pynumdiff/tests/test_diff_methods.py

```diff
@@ -8,15 +8,17 @@
 from ..total_variation_regularization import velocity, acceleration, jerk, iterative_velocity, smooth_acceleration, jerk_sliding
 from ..kalman_smooth import constant_velocity, constant_acceleration, constant_jerk
 from ..smooth_finite_difference import mediandiff, meandiff, gaussiandiff, friedrichsdiff, butterdiff
-# Function alias for testing a case where parameters change the behavior in a big way
+# Function aliases for testing cases where parameters change the behavior in a big way, so error limits can be indexed in dict
 def iterated_second_order(*args, **kwargs): return second_order(*args, **kwargs)
 def iterated_fourth_order(*args, **kwargs): return fourth_order(*args, **kwargs)
+def spline_irreg_step(*args, **kwargs): return splinediff(*args, **kwargs)
 
 dt = 0.1
 t = np.linspace(0, 3, 31) # sample locations, including the endpoint
 tt = np.linspace(0, 3) # full domain, for visualizing denser plots
 np.random.seed(7) # for repeatability of the test, so we don't get random failures
 noise = 0.05*np.random.randn(*t.shape)
+t_irreg = t + np.random.uniform(-dt/3, dt/3, *t.shape) # add jostle
```

Comment:
Thinking through how to test the irregular step case took more thought than actually extending splinediff to do it.